gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -154,17 +177,16 @@ public long getTokenExpiration(final String tokenId) throws UnknownTokenExceptio
 
   @Override
   public long getTokenExpiration(String tokenId, boolean validate) throws UnknownTokenException {
-    long expiration;
-
     if (validate) {
       validateToken(tokenId);
     }
 
+    long expiration = -1;
     synchronized (tokenExpirations) {
-      if (!tokenExpirations.containsKey(tokenId)) {
-        throw new UnknownTokenException(tokenId);
-      }
-      expiration = tokenExpirations.get(tokenId);
+      expiration = tokenExpirations.getOrDefault(tokenId, -1L);
+    }
+    if (expiration == -1) {
```

Comment:
This doesn't appear to be any different than the previous implementation. If this were a check for _expiration < 0_, I could see a difference. What is the reason for the change?

Reply:
There is a slight difference: we can save one iteration when checking `tokenExpirations`
`containsKey + get` vs. `getOrDefault`.
But you are right, there is no more difference between the two versions.

Reply:
I left it as-is based on my previous comment.